bridges/AppleAppStoreBridge.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+class AppleAppStoreBridge extends BridgeAbstract {
+
+	const MAINTAINER = 'captn3m0';
+	const NAME = 'Apple App Store';
+	const URI = 'https://apps.apple.com/';
+	const CACHE_TIMEOUT = 3600; // 1h
+	const DESCRIPTION = 'Returns version updates for a specific application';
+
+	const PARAMETERS = array(array(
+		'id' => array(
+			'name' 			=> 'Application ID',
+			'required' 		=> true,
+			'exampleValue' 	=> '310633997'
+		),
+		'p' => array(
+			'name' 		=> 'Platform',
+			'type' 		=> 'list',
+			'values' 	=> array(
+				'iPad' 		=> 'ipad',
+				'iPhone' 	=> 'iphone',
+				'Mac' 		=> 'mac',
+
+				// The following 2 are present in responses
+				// but not yet tested
+				'Web' 		=> 'web',
+				'Apple TV' 	=> 'appletv',
+			),
+			'defaultValue' => 'iphone',
+		),
+		'country' => array(
+			'name' 			=> 'Store Country',
+			'type' 			=> 'list',
+			'values' 		=> array(
+				'US'	 => 'US',
+				'India'	 => 'IN',
+				'Canada' => 'CA'
+			),
+			'defaultValue' 	=> 'US',
+		),
+	));
+
+	const PLATFORM_MAPPING = array(
+		'iphone' 	=> 'ios',
+		'ipad'		=> 'ios',
+	);
+
+	private function makeHtmlUrl($id, $country){
+		return 'https://apps.apple.com/' . $country . '/app/id' . $id;
+	}
+
+	private function makeJsonUrl($id, $platform, $country){
+		return "https://amp-api.apps.apple.com/v1/catalog/$country/apps/$id?platform=$platform&extend=versionHistory";
+	}
+
+	/**
+	 * In case of some platforms, the data is present in the initial response
+	 */
+	private function getDataFromShoebox($id, $platform, $country){
+		$uri = $this->makeHtmlUrl($id, $country);
+		$html = getSimpleHTMLDOM($uri, 3600);
+		$script = $html->find('script[id="shoebox-ember-data-store"]', 0);
+
+		$json = json_decode($script->innertext, true);
+		return $json['data'];
+	}
+
+	private function getJWTToken($id, $platform, $country){
+		$uri = $this->makeHtmlUrl($id, $country);
+
+		$html = getSimpleHTMLDOM($uri, 3600);
+
+		$meta = $html->find('meta[name="web-experience-app/config/environment"]', 0);
+
+		$json = urldecode($meta->content);
+
+		$json = json_decode($json);
+
+		return $json->MEDIA_API->token;
+	}
+
+	private function getAppData($id, $platform, $country, $token){
+		$uri = $this->makeJsonUrl($id, $platform, $country);
+
+		$headers = [
+			"Authorization: Bearer $token",
+		];
+
+		$json = json_decode(getSimpleHTMLDOM($uri, $headers), true);
```

Comment:
I don't think simplehtmldom is of any use here.

```suggestion
		$json = json_decode(getContents($uri, $headers), true);
```